packages/history-service/src/base-history.ts

```diff
@@ -148,4 +159,32 @@ export abstract class BaseHistory implements ConsumerHistory {
       rootHistoryKey: this.rootHistory.location.key
     };
   }
+
+  private updateRootHistoryKey(oldKey: string): void {
+    const consumerLocation = this.consumerHistory.entries.find(
+      ({state}) => state.rootHistoryKey === oldKey
+    );
+
+    if (consumerLocation) {
+      this.setRootHistoryKey(consumerLocation);
+    }
+  }
+
+  private handleRootLocationChange(
+    location: history.Location,
+    action: history.Action
+  ): void {
+    if (action === 'REPLACE') {
+      /* istanbul ignore else */
+      if (this.currentRootHistoryKey) {
+        this.updateRootHistoryKey(this.currentRootHistoryKey);
```

Comment:
This method should be callable without arguments and should be named `updateAllRootHistoryKeys`